File: 01blog-front/src/app/models/get-posts-request.ts
import { CommentRequest } from "./comment-request";
import { PostImageRequest } from "./post-image-request";
import { Profile } from "./profile-request";
import { User } from "./user";

export interface GetPostsRequest {
    id ?: number;
    title: string;
    content: string;
    category: string;
    creator: Profile;
    dateFrom: string;
    updateDate: string;
    // images: PostImageRequest[];
    // comments: CommentRequest[];
    likes: string[];
    likeCount: number;
    commentsCount: number;
    visibility: string;
}

    //services and components

File: 01blog/src/main/java/com/_Talent/_blog/dto/GetPostsRequest.java
package com._Talent._blog.dto;

import java.util.List;

import lombok.Data;
@Data
public class GetPostsRequest {
    private Long id;
    private String title;
    private String content;
    private String creator;
    private String avatarurl;
    private String dateFrom;
    private String updatedate;
    private List<String> categories;
    private String status;
    private int commentsCount;
    private int likes;
    private int dislikes;
    private boolean likedByCurrentUser;
    private boolean dislikedByCurrentUser;
    private List<String> postLikesusers;
    private List<CommentRequest> comments;
}